File: c#/Lab10/Lab10_1/Circle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab10_1
{
    
    class Circle
    {

        private double radius1;
        public double Radius
        {
            get
            {
                return radius1;
            }
            set
            {
                this.radius1 = value;
                this.type = "round";
            }
        }
        public double Radius1
        {
            get
            {
                return radius1;
            }
            set
            {
                this.radius1 = value;
                this.type = "ellipse";
            }
        }

        private double radius2;
        public double Radius2
        {
            get
            {
                return radius2;
            }
            set
            {
                this.radius2 = value;
            }
        }
        
        private double x { get; set; }
        public double X
        {
            get
            {
                return this.x;
            }
            set
            {
                this.x = value;
            }
        }

        private double y { get; set; }
        public double Y
        {
            get
            {
                return this.y;
            }
            set
            {
                this.y = value;
            }
        }

        private string type;
        public string Type { get { return this.type; } }

        public Circle()
        {

        }

        public Circle(double radius, double x, double y)
        {
            this.radius1 = radius;
            this.x = x;
            this.y = y;
            this.type = "round";
        }

        public Circle(double radius1, double radius2, double x, double y)
        {
            this.radius1 = radius1;
            this.radius2 = radius2;
            this.x = x;
            this.y = y;
            this.type = "ellipse";
        }

        public double GetArea()
        {
            double area;
            if (this.type == "round")
            {
                area = Math.PI * Math.Pow(this.radius1, 2);
            }
            else
            {
                area = Math.PI * this.radius2 * this.radius1;
            }
            return area;
        }

        public double GetCircumference()
        {
            double carcumference;
            if (this.type == "round")
            {
                carcumference = 2 * Math.PI * this.radius1;
            }
            else
            {
                carcumference = Math.PI * (this.radius1 + this.radius2);
            }
            return carcumference;
        }

        public void GetInfo()
        {
            if (this.type == "round")
            {
                Console.WriteLine($"Diameter : {this.radius1 * 2}\nCoordinates : ({this.x}; {this.y})");
            }
            else
            {
                Console.WriteLine($"Diameter1 : {this.radius1 * 2}\nDiameter2 : {this.radius2 * 2}\nCoordinates : ({this.x}; {this.y})");
            }
        }

        public void InputCoordinates()
        {
            double x, y;
            while (true)
            {
                Console.WriteLine("Input x : ");
                if (!double.TryParse(Console.ReadLine(), out x))
                {
                    Console.WriteLine("Value error!");
                }
                else
                {
                    Console.WriteLine("Input y : ");
                    if (!double.TryParse(Console.ReadLine(), out y))
                    {
                        Console.WriteLine("Value error!");
                    }
                    else break;
                }
            }
            this.x = x;
            this.y = y;
        }
    }
}

File: python/Lab10/circle.py
import math

def correct_float(string):
    while True:
        try:
            point = float(input("{}".format(string)))
        except ValueError:
            print("Value error! Try again!\n")
        else:
            if point >= 0:
                return point
            print("Range error! Try again!\n")

class Circle(object):    
    def __init__(self, x = None, y = None, radius1 = None, radius2 = None) -> None:
        self.coord = (x, y)
        self.radius1 = radius1
        if radius2 == None:
            self.type = "round"
        else:
            self.radius2 = radius2
            self.type = "ellipse"
    
    @property
    def set_coord(self):
        x = correct_float("Input x : ")
        y = correct_float("Input y : ")
        self.coord = (x, y)
    @property
    def get_coord(self):
        return self.coord
    @property
    def get_type(self):
        return self.type
    
    
    
    def get_area(self):
        area = 0.0
        if self.type == "round":
            area = math.pi * (self.radius1 ** 2)
        else:
            area = math.pi * self.radius1 * self.radius2
        return area
    
    def get_circumfence(self):
        circumfence = 0.0
        if self.type == "round":
            circumfence = 2 * math.pi * self.radius1
        else:
            circumfence = math.pi * (self.radius1 + self.radius2)
        return circumfence
    
    def get_info(self):
        if self.type == "round":
            print(f"Diameter : {self.radius1 * 2}\nCoordinates : {self.coord}")
        else:
            print(f"Diameter1 : {self.radius1 * 2}\nDiameter2 : {self.radius2 * 2}\nCoordinates : {self.coord}")
